### ISSUE TITLE
External model inference

### ISSUE THREAD

**Issue by sheridana** [talmolab/sleap#1119]
# Problem background

* There are various pretrained pose estimation models that could be integrated into SLEAP. It would be nice to provide a scaffolding to support these models. 
* A good example to start with is [MoveNet](https://blog.tensorflow.org/2021/05/next-generation-pose-detection-with-movenet-and-tensorflowjs.html) - a pose estimation model from Google. 
* This would provide the added benefit of support for human pose estimation in SLEAP.
* This would also provide an example for supporting models from [tensorflow hub](https://www.tensorflow.org/hub)

# Feature proposal

* Add support for inference with pre-trained MoveNet model(s).

# Implementation details

### Example MoveNet inference
* [Here](https://github.com/talmolab/movenet-dance-demo) is an example use of MoveNet for real-time pose matching
* This example has a wrapper function to [load pretrained models](https://github.com/talmolab/movenet-dance-demo/blob/54eb79cb0addf0b5e9e5a3b49ea3eea07ff6ed06/inference.py#L40) which returns a [keras model](https://github.com/talmolab/movenet-dance-demo/blob/54eb79cb0addf0b5e9e5a3b49ea3eea07ff6ed06/inference.py#L72)
* The keras model can then run inference by calling `predict_on_batch`, handled [here](https://github.com/talmolab/movenet-dance-demo/blob/54eb79cb0addf0b5e9e5a3b49ea3eea07ff6ed06/inference.py#L97) for a single image

### Corresponding SLEAP code
* The corresponding logic to load pretrained models in SLEAP is handled inside the predictor `from_trained_models` class method ([example](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L1421))
* This instantiates an `InferenceModel` (subclasses keras model) which also handles the [prediction](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L954)
* At a high level, the base `Predictor` class handles prediction via a [generator](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L444) which first takes care of [data preprocessing](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L319) and then runs [inference](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L325)

### This PR
* Since there is a lot of specific logic in current predictor classes, it is probably best to create a separate predictor (e.g `MoveNetPredictor`) that subclasses the base `Predictor` class. 
* This predictor will need to be exposed, similar to what is done for the [other predictors](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L221)
* The `MoveNetPredictor` should  handle loading from tensorflow hub (rather than via `tf.keras.models.load_model`) as was done in the example
* We probably will want to override the [pipeline creation](https://github.com/talmolab/sleap/blob/9ae2941649c81ca5b8e8301ccd99725732599fa5/sleap/nn/inference.py#L254), to support center padding as done in the example (or maybe this is already fine with the `SizeMatcher`?)
* It seems like MoveNet expects a batch size of 1, so this would need to be set appropriately in the predictor

**Comment by roomrys** [comment]
The PRs to integrate MoveNet have been merged; however, we still need to expose the MoveNet option to the GUI (and add support in the configuration files).

**Comment by roomrys** [comment]
This feature is now available in the (pre) release 1.3.0a0, to install, first [uninstall](https://sleap.ai/installation.html#upgrading-and-uninstalling) and then:
**`conda` (Windows/Linux/GPU)**:
```
conda create -y -n sleap -c sleap -c sleap/label/dev -c nvidia -c conda-forge sleap=1.3.0a0
```
**`pip` (any OS except Apple Silicon)**:
```
pip install sleap==1.3.0a0
```

**Warning:** This is a pre-release! Expect bugs and strange behavior when testing.
